.github/workflows/linux.yml

```diff
@@ -8,13 +8,49 @@ jobs:
     strategy:
       matrix:
         cxx: [g++-4.8, g++-8, g++-10, clang++-9]
+        standard: [11, 14, 17, 20]
         build_type: [Debug, Release]
         include:
           - cxx: clang++-9
             build_type: Debug
             fuzz: -DFMT_FUZZ=ON -DFMT_FUZZ_LINKMAIN=ON
           - cxx: g++-4.8
             install: sudo apt install g++-4.8
+        exclude: # filter some configurations
+          ### gcc 4.8
+          # does not fully support c++14
+          - cxx: g++-4.8
+            standard: 14
+          # does not support c++17
+          - cxx: g++-4.8
+            standard: 17
+          # does not support c++20
+          - cxx: g++-4.8
+            standard: 20
+
+          ### gcc 8
+          # has c++14 as default mode
```

Comment:
Doesn't mean someone won't use c++11 on gcc-8.
If resources aren't a constrain I wouldn't exclude it from checks.

Same comment for other excludes like this one.

Reply:
I was just impressed by the number of configurations without any filtering.
Maybe I over-filtered them. 
But I don't know if they make any sense, for example, is it possible that c++11 on gcc-8 fails while c++11 on gcc-4.8 doesn't? It can be possible of course, because of some bugs, but if we are hunting for bugs here then this config should have a huge number of configurations.

Reply:
I see your point.
However following this though, standard shouldn't be a matrix parameter.
Check default on each compiler and have separate matrix for all cpp standards tested on newest compilers only.

Reply:
Ok, this is a good suggestion.

Reply:
But I think it will be better to create a new PR with the described logic because the logic here and the logic you proposed differ. I can create it, or maybe you want to do this?